server/xpub-controller/manuscript.js

```diff
@@ -99,84 +73,36 @@ class Manuscript {
       )
       await fileEntity.updateStatus('CANCELLED')
       await fileEntity.delete()
-      clearInterval(handle)
+      FilesHelper.endFileProgress(progress)
       throw err
     }
 
     logger.info(`Manuscript Upload S3::end ${filename} | ${manuscriptId}`)
 
-    let title = ''
-    try {
-      // also send source file to conversion service
-      title = await this.scienceBeamApi.extractSemantics(
-        this.config,
-        fileContents,
-        filename,
-        mimeType,
-      )
-    } catch (error) {
-      let errorMessage = ''
-      if (error.error.code === 'ETIMEDOUT' || error.error.connect === false) {
-        errorMessage = 'Request to science beam timed out'
-      } else {
-        errorMessage = error.message
-      }
-      logger.warn('Manuscript conversion failed', {
-        error: errorMessage,
-        manuscriptId,
-        filename,
-      })
-    }
-
-    // After the length file operations above - now update the manuscript...
-    manuscript = await ManuscriptModel.find(manuscriptId, this.userId)
-    const oldFileIndex = manuscript.files.findIndex(
-      element => element.type === 'MANUSCRIPT_SOURCE',
-    )
-
-    logger.info(
-      `Manuscript Upload found index ${oldFileIndex} ${
-        fileEntity.filename
-      } | ${manuscriptId}`,
-    )
-
-    if (oldFileIndex >= 0) {
-      const oldFile = await FileModel.find(manuscript.files[oldFileIndex].id)
-      manuscript.files.splice(oldFileIndex, 1)
-      await oldFile.delete()
-    }
-
-    const pendingFileIndex = manuscript.files.findIndex(
-      element => element.type === 'MANUSCRIPT_SOURCE_PENDING',
+    const title = await this.filesHelper.extractFileTitle(
+      fileContent,
+      filename,
+      mimeType,
+      manuscriptId,
     )
+    manuscript = await ManuscriptModel.find(manuscriptId, this.userId)
+    FilesHelper.cleanOldManuscript(manuscript)
 
-    manuscript.files[pendingFileIndex].type = 'MANUSCRIPT_SOURCE'
-
-    logger.info(`Manuscript Upload Manuscript::save ${title} | ${manuscriptId}`)
-    manuscript.meta.title = title
-    await manuscript.save()
+    manuscript = await FilesHelper.setManuscriptMetadata(manuscript, title)
 
-    const sourceList = manuscript.files.filter(
-      f => f.type === 'MANUSCRIPT_SOURCE',
-    )
-    const pendingList = manuscript.files.filter(
-      f => f.type === 'MANUSCRIPT_SOURCE_PENDING',
-    )
-    if (sourceList.length !== 1 || pendingList.length !== 0) {
-      logger.error(`Validation failed ${JSON.stringify(manuscript, null, 4)}`)
-      throw new Error(`Validation Failure on ${manuscript.id}`)
-    }
+    FilesHelper.validateManuscriptSource(manuscript)
 
     logger.info(
       `Manuscript Upload Manuscript::saved ${
         manuscript.meta.title
       } | ${manuscriptId}`,
     )
 
-    clearInterval(handle)
+    FilesHelper.endFileProgress(progress)
     pubsub.publish(`${ON_UPLOAD_PROGRESS}.${manuscriptId}`, {
```

Comment:
Could `endFileProgress` be mixed with the publish 100%? I don't think we should be calling one without the other anywhere.